### PR TITLE
ESCP-1293:  Strongly typed TimeCodeRange and TimeCode

### PR DESCRIPTION
**Pull request by shabtaisharon**

### CODE REVIEW

**Comment by rpmoore** [comment]
Conflict

**Comment by shabtaisharon** [comment]
> Conflict

fixed

File: SpectraLogic.SpectraRioBrokerClient/Exceptions/MixTimeCodeRangeException.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ///
+    /// </summary>
+    /// <seealso cref="System.Exception" />
+    public class MixTimeCodeException : Exception
+    {
+        #region Constructors
+
+        /// <summary>Initializes a new instance of the <see cref="MixTimeCodeException"/> class.</summary>
+        /// <param name="message">The message that describes the error.</param>
+        public MixTimeCodeException(string message) : base(message)
```
Comment (rpmoore) [inline SpectraLogic.SpectraRioBrokerClient/Exceptions/MixTimeCodeRangeException.cs]: What does a `MixTimeCodeException` mean?

Comment (shabtaisharon) [reply SpectraLogic.SpectraRioBrokerClient/Exceptions/MixTimeCodeRangeException.cs]: mix of drop and non-drop frame in the given range

Comment (rpmoore) [reply SpectraLogic.SpectraRioBrokerClient/Exceptions/MixTimeCodeRangeException.cs]: Add that to the documentation for this exception.

Comment (shabtaisharon) [reply SpectraLogic.SpectraRioBrokerClient/Exceptions/MixTimeCodeRangeException.cs]: done

**Pull request merged**